### DISCUSSION TITLE
Imported GLB 'too large to generate a floor plan'

### DISCUSSION THREAD

**Discussion by opera10** [mozilla/hubs#4795] [Spoke Questions] [answered]
Hey everyone,
I'm currently working on a project in Spoke/Hubs for what I edited a 3D model in Blender. Trying to upload the GLB file leads to the attached error notification. Unchecking the 'walkable" checkbox worked so far, but I actually need that object to be walkable. So far, I've scaled down the model in Blender and croped the top as well - unfortunately that did not help at all - I still recieve that error message. Anybody out there who might help?
![Bild 1](https://user-images.githubusercontent.com/93386398/139438133-d00c747f-e305-48b0-802d-b2fb124bac0d.png)

Nevertheless thanks for your great work, keep it up!

Thanks alot and best wishes
Opera10

**Comment by j-conrad** [comment, marked as answer]
When dealing with large or complex models, it's sometimes a better option to use the option inside the Floor Plan element that lets you point it toward a custom mesh that you import. This custom mesh can be made in any 3d software (like Blender) and is often made by selecting the floor meshes, duplicating the appropriate faces, and then making them their own object. This avoids the Floor Plan's built-in calculation and just uses your custom mesh exactly as it appears. For large scenes, this is sometimes the only way to avoid those errors.

Link to a gif showing the UI:
https://gyazo.com/0319ae14c039625b1c7ad5ea1d897d61

**Reply by opera10** [reply]
**Thank you!**
